torch/_inductor/kernel/bmm.py

```diff
@@ -178,8 +189,14 @@ def may_require_contiguous(t, meta_t):
         layout,
     )
 
+    if out_dtype:
+        assert mat1.get_device().type == "cuda", "out_dtype is only supported for CUDA"
```

Comment:
Hi, may I suggest you generalize this so that XPU can benefit from this?
`assert is_gpu(mat1.get_device().type)`